reactor-core/src/main/java/reactor/core/scheduler/BoundedElasticScheduler.java

```diff
@@ -160,61 +146,159 @@ BoundedScheduledExecutorService createBoundedExecutorService() {
 
 	@Override
 	public boolean isDisposed() {
-		return BOUNDED_SERVICES.get(this) == SHUTDOWN;
+		return state.currentResource.boundedServices == SHUTDOWN;
 	}
 
 	@Override
 	public void start() {
+		SchedulerState<BoundedServicesState> a = this.state;
+		if (a.currentResource.boundedServices != SHUTDOWN
+				&& a.currentResource.boundedServices != SHUTTING_DOWN) {
+			return;
+		}
+		SchedulerState<BoundedServicesState> b = SchedulerState.init(
+				BoundedServicesState.wrap(new BoundedServices(this))
+		);
 		for (;;) {
-			BoundedServices services = BOUNDED_SERVICES.get(this);
-			if (services != SHUTDOWN) {
-				return;
+			a = this.state;
+			if (a.currentResource.boundedServices == SHUTTING_DOWN) {
+				// keep spinning until previous shuts down
+				continue;
 			}
```

Comment:
SHUTTING_DOWN should not be a reason to continue spinning. We have to proceed and set a new scheduler or return immediately